bugsnag-android-core/src/main/java/com/bugsnag/android/FileStore.java

```diff
@@ -137,6 +145,26 @@ void discardOldestFileIfNeeded() {
     @NonNull
     abstract String getFilename(Object object);
 
+    /**
+     * Determines whether a given file is a tombstone. Tombstones were added on API 26 - a
+     * zero-length file is left behind rather than deleting the file from the cache dir entirely.
+     *
+     * See https://developer.android.com/reference/android/os/storage/StorageManager.html
+     * @param file the cached file
+     * @return true if the file is a tombstone
+     */
+    private boolean isTombstone(File file) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            try {
+                return file.length() == 0 && storageManager.isCacheBehaviorTombstone(cacheDir);
```

Comment:
Would this value be true if `cacheDir` + `/bugsnag-errors` is using tombstone behavior but `cacheDir` is not? Can `path` be checked directly?

Reply:
Updated to check the path directly.